### PR TITLE
[15.0][ADD] stock_picking_auto_print: New module

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @Tecnativa TT51076

ping @chienandalu

### CODE REVIEW

File: stock_picking_auto_print/readme/DESCRIPTION.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+This module allows to auto print delivery slip after validation
```
Comment (chienandalu) [inline stock_picking_auto_print/readme/DESCRIPTION.rst]:
```suggestion
This is a backport of the Odoo 17 functionality that allows to auto print delivery slip after validation
```

File: stock_picking_auto_print/models/stock_picking_type.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Copyright 2024 Tecnativa - Sergio Teruel
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class StockPickingType(models.Model):
+    _inherit = "stock.picking.type"
+
+    auto_print_delivery_slip = fields.Boolean(
```
Comment (chienandalu) [inline stock_picking_auto_print/models/stock_picking_type.py:10]: Don't we add the [`auto_print_return_slip`](https://github.com/OCA/OCB/blob/381c3681b55d185e2b27e8001edafe0dc284d6b4/addons/stock/models/stock_picking.py#L87) as well?

File: stock_picking_auto_print/models/stock_picking.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+_logger = logging.getLogger(__name__)
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    def button_validate(self):
+        res = super().button_validate()
+
+        pickings_to_print = self.filtered(
```
Comment (chienandalu) [inline stock_picking_auto_print/models/stock_picking.py:47]: Why not implementing the original `_get_autoprint_report_actions` hook method?: https://github.com/OCA/OCB/blob/381c3681b55d185e2b27e8001edafe0dc284d6b4/addons/stock/models/stock_picking.py#L1658-L1665

File: stock_picking_auto_print/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# Copyright 2024 Tecnativa - Sergio Teruel
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+{
+    "name": "Stock Picking Auto Print",
+    "summary": "Print picking delivery slip automatically after validation",
+    "version": "15.0.1.0.0",
+    "author": "Tecnativa, Odoo Community Association (OCA)",
```
Comment (chienandalu) [inline stock_picking_auto_print/__manifest__.py:8]:
```suggestion
    "author": "Odoo S.A., Tecnativa, Odoo Community Association (OCA)",
```

File: stock_picking_auto_print/__manifest__.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
+# Copyright 2024 Tecnativa - Sergio Teruel
```
Comment (chienandalu) [inline stock_picking_auto_print/__manifest__.py:1]:
```suggestion
# Copyright 2024 Tecnativa - Sergio Teruel
# Copyright 2024 Odoo S.A.
```

File: stock_picking_auto_print/models/stock_picking_type.py (hunk 2 above)
Comment (sergio-teruel) [reply stock_picking_auto_print/models/stock_picking_type.py:10]: In v15 does not exist the returned report

File: stock_picking_auto_print/views/stock_picking_type_views.xml [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+    <record model="ir.ui.view" id="view_partner_valued_picking">
+        <field name="model">stock.picking.type</field>
+        <field name="inherit_id" ref="stock.view_picking_type_form" />
+        <field name="arch" type="xml">
+            <sheet>
+                <group name="auto_print">
+                    <group string="Print on Validation">
+                        <field name="auto_print_delivery_slip" string="Delivery Slip" />
+                        <field name="auto_print_return_slip" string="Return Slip" />
```
Comment (chienandalu) [inline stock_picking_auto_print/views/stock_picking_type_views.xml]: This doesn't exist
```suggestion
```

File: stock_picking_auto_print/models/stock_picking_type.py (hunk 2 above)
Comment (chienandalu) [reply stock_picking_auto_print/models/stock_picking_type.py:10]: Ah, ok :+1:

**Comment by sergio-teruel** [comment]
Changes done!

**Comment by sergio-teruel** [comment]
@chienandalu We use the module https://github.com/OCA/web/tree/15.0/web_ir_actions_act_multi

ping @pedrobaeza

**Review by chienandalu — COMMENTED** [review]
Tests failing now...

**Comment by sergio-teruel** [comment]
@chienandalu Fixed!!

**Review by chienandalu — APPROVED** [review]
Working right :+1:

File: stock_picking_auto_print/views/stock_picking_type_views.xml [hunk 7]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+
+    <record model="ir.ui.view" id="view_partner_valued_picking">
```
Comment (chienandalu) [inline stock_picking_auto_print/views/stock_picking_type_views.xml]:
```suggestion
    <record model="ir.ui.view" id="view_picking_type_form">
```

**Review by chienandalu — CHANGES_REQUESTED** [review]

File: stock_picking_auto_print/views/stock_picking_type_views.xml [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+        <field name="model">stock.picking.type</field>
+        <field name="inherit_id" ref="stock.view_picking_type_form" />
+        <field name="arch" type="xml">
+            <sheet>
+                <group name="auto_print">
+                    <group string="Print on Validation">
```
Comment (chienandalu) [inline stock_picking_auto_print/views/stock_picking_type_views.xml]:
```suggestion
                    <group name="print_on_validation" string="Print on Validation">
```

**Comment by sergio-teruel** [comment]
Changes done!!

**Comment by sebalix** [comment]
Not exactly the same (and here it's a backport of existing std solution), but putting the link of a similar solution done in v14 based on `base_report_to_printer`: https://github.com/OCA/report-print-send/pull/310 (not merged)

**Comment by chienandalu** [comment]
Hi @sebalix , thank for the heads up. We're backporting Odoo 17.0 stuff here so we reduce dependencies once we move to that version and beyond

**Review by chienandalu — APPROVED** [review]

File: stock_picking_auto_print/__manifest__.py (hunk 4 above)
Comment (pedrobaeza) [reply stock_picking_auto_print/__manifest__.py:8]: I see that the code is different enough, and the core one doesn't include tests, that we can skip this attribution. Do you agree?

Comment (chienandalu) [reply stock_picking_auto_print/__manifest__.py:8]: Ok, it's true that although the API will be the same and no migration would work be needed in v17, the code adaptations had to be deep

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge nobump

**Pull request merged**